### PR TITLE
Update samples and test sites to new formats

### PR DESCRIPTION
**Pull request by ryanbrandenburg**

### CODE REVIEW

File: samples/CustomRouteSample.Web/project.json [hunk 1]
```diff
@@ -10,11 +10,12 @@
...
   },
   "frameworks": {
-    "net451": {},
+    "net451": { },
     "netcoreapp1.0": {
       "imports": [
         "dnxcore50",
-        "portable-net451+win8"
+        "portable-net451+win8",
+
```
Comment (dougbu) [inline samples/CustomRouteSample.Web/project.json]: Nah, no trailing comma if you want valid JSON. (Don't need the blank line either.)

File: samples/EmbeddedViewSample.Web/project.json [hunk 2]
```diff
@@ -12,11 +12,12 @@
...
   },
   "frameworks": {
-    "net451": {},
+    "net451": { },
     "netcoreapp1.0": {
       "imports": [
         "dnxcore50",
-        "portable-net451+win8"
+        "portable-net451+win8",
+
```
Comment (dougbu) [inline samples/EmbeddedViewSample.Web/project.json]: What? Does this actually build?

File: test/Microsoft.AspNetCore.Mvc.Abstractions.Test/project.json [hunk 3]
```diff
@@ -13,6 +13,7 @@
     "netcoreapp1.0": {
       "imports": [
         "dnxcore50",
+         ,
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Abstractions.Test/project.json]: What?

File: test/WebSites/BasicWebSite/project.json [hunk 4]
```diff
@@ -31,5 +31,13 @@
   "content": [
     "Views",
     "web.config"
```
Comment (dougbu) [inline test/WebSites/BasicWebSite/project.json]: Add `"Areas/Area1/Views"` to this array.

**Comment by dougbu** [comment]
:watch:

File: test/Microsoft.AspNetCore.Mvc.Abstractions.Test/project.json (hunk 3 above)
Comment (ryanbrandenburg) [inline test/Microsoft.AspNetCore.Mvc.Abstractions.Test/project.json]: Looking back at my terminal history it looks like after I did a find and replace I tried to do a build to check that it all worked, but built the wrong repo without noticing. Cleaning all these up now.

**Comment by ryanbrandenburg** [comment]
:up::date:

**Comment by dougbu** [comment]
1. Set content root
2. Make sure `web.config` and `*.cshtml` files are published
3. Add missing Razor tooling in a few cases

Did I miss anything? If that's the story, LGTM.

**Comment by dougbu** [comment]
:shipit:

**Comment by ryanbrandenburg** [comment]
Yep, that's the idea, will rebase and merge.

**Pull request merged**